AssertionError:

Crash Report
============

This crash report was reported through the automatic crash reporting system 🤖

Traceback
--------------

```Python traceback
Traceback (most recent call last):
  File "electrum\gui\qt\send_tab.py", line 571, in do_pay_or_get_invoice
  File "electrum\gui\qt\send_tab.py", line 518, in read_invoice
  File "electrum\payment_identifier.py", line 682, in invoice_from_payment_identifier
AssertionError

AssertionError: 
```

Reporter
------------

This issue was reported by 1 user(s):

| Electrum Version  | Python Version | Operating System  | Wallet Type  | Locale | Stack |
|---|---|---|---|---|---|
| 4.5.3  | 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:20:04) [MSC v.1929 32 bit (Intel)] | Windows-10-10.0.22631-SP0 | standard | en_US | [ℹ](https://crashhub.electrum.org//crash/31866) | 


Additional Information
------------------------

The reporting user(s) did not provide additional information.

--- a/electrum/gui/qt/send_tab.py
+++ b/electrum/gui/qt/send_tab.py
@@ -242,9 +242,11 @@ class SendTab(QWidget, MessageBoxMixin, Logger):
 
         if pi is None or pi.type == PaymentIdentifierType.UNKNOWN:
             return
-
-        assert pi.type in [PaymentIdentifierType.SPK, PaymentIdentifierType.MULTILINE,
-                           PaymentIdentifierType.BIP21, PaymentIdentifierType.OPENALIAS]
+        elif pi.type not in [PaymentIdentifierType.SPK, PaymentIdentifierType.MULTILINE,
+                           PaymentIdentifierType.BIP21, PaymentIdentifierType.OPENALIAS]:
+            # clear the amount field once it is clear this PI is not eligible for '!'
+            self.amount_e.clear()
+            return
 
         if pi.type == PaymentIdentifierType.BIP21:
             assert 'amount' not in pi.bip21
@@ -406,6 +408,8 @@ class SendTab(QWidget, MessageBoxMixin, Logger):
             self.amount_e.setFrozen(lock_amount)
         if lock_max is not None:
             self.max_button.setEnabled(not lock_max)
+            if lock_max is True:
+                self.max_button.setChecked(False)
         if lock_description is not None:
             self.message_e.setFrozen(lock_description)
 
@@ -466,6 +470,8 @@ class SendTab(QWidget, MessageBoxMixin, Logger):
         # resolve '!' in amount editor if it was set before PI
         if not lock_max and self.amount_e.text() == '!':
             self.spend_max()
+        elif lock_max and self.amount_e.text() == '!':
+            self.amount_e.clear()
 
         pi_unusable = pi.is_error() or (not self.wallet.has_lightning() and not pi.is_onchain())
         is_spk_script = pi.type == PaymentIdentifierType.SPK and not pi.spk_is_address

